providers/flagd/pkg/service/service.go

```diff
@@ -0,0 +1,179 @@
+package service
+
+import (
+	"errors"
+
+	"github.com/bufbuild/connect-go"
+	"github.com/open-feature/go-sdk/pkg/openfeature"
+	"golang.org/x/net/context"
+
+	flagdModels "github.com/open-feature/flagd/pkg/model"
+	log "github.com/sirupsen/logrus"
+	schemaV1 "go.buf.build/open-feature/flagd-connect/open-feature/flagd/schema/v1"
+	"google.golang.org/protobuf/types/known/structpb"
+)
+
+type ServiceConfiguration struct {
+	Port            uint16
+	Host            string
+	CertificatePath string
+	SocketPath      string
+}
```

Comment:
Do we need a boolean for `Tls`?

I would assume that if the remote flagd has a cert signed by a popular CA, default system certs would be used, so there'd be no need for a `CertificatePath`. In that case, if you want to use TLS, what do you do? Or am I wrong and you always need to provide certs? That would surprise me.

Reply:
see https://github.com/open-feature/js-sdk-contrib/blob/321ac0936c6744c6cf9956c6708a701a3d9237be/libs/providers/flagd/src/lib/configuration.spec.ts#L25-L28